### PR TITLE
Optical circuits

### PR DESCRIPTION
**Pull request by pablocnvs**
Optical circuits, some more sensor and emmiter components and wire insulation for higher tiers

### CODE REVIEW

File: src/main/java/gregicadditions/GAMaterials.java [hunk 1]
```diff
@@ -1080,7 +1233,7 @@
...
     public static final IngotMaterial UVSuperconductor = new IngotMaterial(744, "uv_superconductor", 0xe0d207, MaterialIconSet.SHINY, 1, of(new MaterialStack(UVSuperconductorBase, 1), new MaterialStack(Nitrogen, 1)), DISABLE_DECOMPOSITION);
-    public static final IngotMaterial UHVSuperconductorBase = new IngotMaterial(740, "uhv_superconductor_base", 0x359ffc, MaterialIconSet.SHINY, 1, of(new SimpleDustMaterialStack(TBCCODust, 1), new SimpleDustMaterialStack(StrontiumSuperconductorDust, 1), new MaterialStack(TriniumTitanium, 2)), STD_METAL, null, 10000);
+    public static final IngotMaterial UHVSuperconductorBase = new IngotMaterial(740, "uhv_superconductor_base", 0x359ffc, MaterialIconSet.SHINY, 1, of(new SimpleDustMaterialStack(TBCCODust, 1), new SimpleDustMaterialStack(StrontiumSuperconductorDust, 1), new MaterialStack(Taranium, 2)), STD_METAL, null, 10000);
```
Comment (hjae78) [inline src/main/java/gregicadditions/GAMaterials.java:1236]: Taranium is really expensive, could you change Taranium to be 1 and the others to be 4 or 5?

File: src/main/java/gregicadditions/recipes/GARecipeAddition.java [hunk 2]
```diff
@@ -766,16 +767,16 @@ public static void init2() {
...
-        ASSEMBLY_LINE_RECIPES.recipeBuilder().duration(600).EUt(245760).outputs(SENSOR_UHV.getStackForm()).inputs(OreDictUnifier.get(frameGt, HDCS, 1), SENSOR_UV.getStackForm(2), LeadScandiumTantalate.getItemStack(64), OreDictUnifier.get(foil, Osmiridium, 64), OreDictUnifier.get(foil, Osmiridium, 64)).input(cableGtSingle, TungstenTitaniumCarbide, 8).inputs(OreDictUnifier.get(gemExquisite, Diamond, 2)).input(circuit, Tier.Ultimate, 8).fluidInputs(SolderingAlloy.getFluid(576)).buildAndRegister();
+        ASSEMBLY_LINE_RECIPES.recipeBuilder().duration(600).EUt(245760).outputs(SENSOR_UHV.getStackForm()).inputs(OreDictUnifier.get(frameGt, HDCS, 1), SENSOR_UV.getStackForm(2), LeadScandiumTantalate.getItemStack(64), OreDictUnifier.get(foil, Osmiridium, 64), OreDictUnifier.get(foil, Osmiridium, 64)).input(cableGtSingle, TungstenTitaniumCarbide, 8).inputs(OreDictUnifier.get(gemExquisite, Diamond, 2)).inputs(BOSE_EINSTEIN_COOLING_CONTAINER.getStackForm(8)).input(circuit, Tier.Ultimate, 8).fluidInputs(SolderingAlloy.getFluid(576)).buildAndRegister();
```
Comment (hjae78) [inline src/main/java/gregicadditions/recipes/GARecipeAddition.java:779]: Maybe use 16 in the UEV recipe?

File: src/main/java/gregicadditions/recipes/chain/SensorEmmiter.java [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+import static gregtech.api.unification.ore.OrePrefix.dust;
+import static gregtech.api.unification.ore.OrePrefix.wireFine;
+
+public class SensorEmmiter {
+    public static void init() {
+        CHEMICAL_RECIPES.recipeBuilder().duration(420).EUt(720)
+                .input(dust, Strontium)
+                .input(dust, Barium)
+                .fluidInputs(AceticAcid.getFluid(2000))
+                .fluidOutputs(BariumStrontiumAcetateSolution.getFluid(2000))
```
Comment (hjae78) [inline src/main/java/gregicadditions/recipes/chain/SensorEmmiter.java:20]: Try to make sure that number of inputs = number of outputs, so here we should have 4000 in the output

**Review by hjae78 — CHANGES_REQUESTED** [review]

**Comment by hjae78** [comment]
Fantastic work as always! Requested some changes

File: src/main/java/gregicadditions/recipes/GARecipeAddition.java (hunk 2 above)
Comment (pablocnvs) [reply src/main/java/gregicadditions/recipes/GARecipeAddition.java:779]: do i move them from UHV to UEV or keep them in UHV and add them in UEV?

Comment (hjae78) [reply src/main/java/gregicadditions/recipes/GARecipeAddition.java:779]: I'd like to see them in both recipes! No reason to not have them in the higher tier component as well

Comment (pablocnvs) [reply src/main/java/gregicadditions/recipes/GARecipeAddition.java:779]: i had put it in both, so nice

**Pull request merged**